dcos-log/mesos/files/reader/read.go

```diff
@@ -349,9 +369,68 @@ start:
 	return strings.NewReader(rm.formatFn(*line, rm)).Read(b)
 }
 
-// Sandbox returns a path to mesos sandbox.
-func (rm ReadManager) Sandbox() string {
-	return rm.sandboxPath
+// SandboxFile represents a file object located in mesos sandbox.
+type SandboxFile struct {
+	GID   string `json:"gid"`
+	Mode  string `json:"mode"`
+	MTime mTime  `json:"mtime"`
+	NLink uint   `json:"nlink"`
+	Path  string `json:"path"`
+	Size  uint64 `json:"size"`
+	UID   string `json:"uid"`
+}
+
+type mTime uint64
+
+// mtime field has a weird format with trailing .0
+// in order to unmarshal this value into json, we need to cut off the suffix .0 if it exists
+func (mt *mTime) UnmarshalJSON(data []byte) error {
+	sanitizedData := bytes.TrimSuffix(data, []byte(".0"))
+	v, err := strconv.ParseUint(string(sanitizedData), 10, 64)
+	if err != nil {
+		return err
+	}
+	*mt = mTime(v)
+	return nil
+}
+
+// BrowseSandbox returns a url to browse files in the sandbox.
+func (rm ReadManager) BrowseSandbox() ([]SandboxFile, error) {
+	v := url.Values{}
+	v.Add("path", rm.sandboxPath)
+
+	newURL := rm.readEndpoint
+	newURL.RawQuery = v.Encode()
+
+	resp, err := rm.client.Get(newURL.String())
+	if err != nil {
+		return nil, fmt.Errorf("unable to make a GET request: %s. URL %s", err, newURL.String())
+	}
+	defer resp.Body.Close()
+
+	var files []SandboxFile
+
+	if err := json.NewDecoder(resp.Body).Decode(&files); err != nil {
+		return nil, fmt.Errorf("unable to decode files API response: %s. URL %s", err, newURL.String())
```

Comment:
you can replace `newURL.String()` here with just `newURL` since you're using `%s`